### PR TITLE
CDAP-17954: add dynamic hiding overflow values for multiSelect

### PR DESCRIPTION
**Pull request by sumengwang**
# CDAP-17954: add dynamic hiding overflow values for multiSelect

## Description
Changed the logic of how MultiSelect renders values. Before it shows <firstElement> + <totalcount - 1>. Now it will try to fill out the available space then show the remaining count number (see attached screenshots)

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-17954](https://cdap.atlassian.net/browse/CDAP-17954)

## Test Plan
Manual test

## Screenshots
image
image

### CODE REVIEW

**Comment by bdmogal** [comment]
@sumengwang tiny feedback: Is there anyway to make the count not look like one of the options? Also, instead of ``+10``, can we say ``10 more``, and have it be a link that opens a tooltip with the other selections?

**Comment by sumengwang** [comment]
@bdmogal I'll incorporate your feedbacks. Thanks

**Comment by sumengwang** [comment]
I changed the selections style from text to cards, and differentiate "...n more" with different color, so that users won't think it's a header.
image

**Comment by sumengwang** [comment]
changed 'n more' to a blue hyperlink style, when hovering on it, it will show a tooltip containing the hidden selections. 

image

File: app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx [hunk 1]
```diff
@@ -42,9 +39,26 @@ const styles = (theme) => {
...
+      backgroundColor: 'white',
+    },
+    hyperlink: {
+      margin: 2,
+      color: '#0000EE',
+    },
   };
 };
 
+const LightTooltip = withStyles((theme) => ({
```
Comment (seanfeld) [inline app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: Can you do this with a styled component instead?

File: app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx [hunk 2]
```diff
@@ -90,23 +120,40 @@ function MultiSelectBase({
...
     }
-    return `${selectionLabel} ${additionalSelectionCount}`;
+    return (
+      <Box>
+        {shownSelections.map((value) => (
+          <Chip className={classes.chip} variant="outlined" key={value} label={value} />
+        ))}
+        {additionalSelectionCount != '' && (
+          <LightTooltip title={additionalSelectionText} placement="right-start">
+            <u className={classes.hyperlink}>{additionalSelectionCount}</u>
```
Comment (seanfeld) [inline app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: I'm not sure this is the correct usage for <u> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/u - is this just to apply an underline styling? If so this should be a <span>.

**Review by seanfeld — CHANGES_REQUESTED** [review]
Check the comment about <u> and then for _new_ styling we should be using styled components.

File: app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx (hunk 2 above)
Comment (sumengwang) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: Yes the u tag is just to provide a underline

File: app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx (hunk 1 above)
Comment (sumengwang) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: Having trouble to use styled component to override Tooltip styles

Comment (mrahanjam) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: What if you use something like:
```javascript
export const StyledTooltip = styled(Tooltip)`
    background-color: ${(props) => props.theme.palette.white},
    color: rgba(0, 0, 0, 0.87),
    box-shadow: ${(props) => props.theme.shadows[1]},
    font-size: 11,
`
```

Comment (seanfeld) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: i think he was saying he couldn't get any of the styles to render with tooltip - i think he needs to do something like this
```
const GroupsContainer = styled.div`
  && {
    && {
      && {
        height: auto;
        overflow: scroll;
        position: absolute;
        right: 1px;
      }
    }
  }
`;
```

Comment (seanfeld) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: ^ specificity issue

Comment (sumengwang) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: Yea, I was only able to override the Tooltip styles with 'withStyles'

File: app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx [hunk 3]
```diff
@@ -42,6 +38,20 @@ const styles = (theme) => {
     root: {
       margin: theme.Spacing(2),
     },
+    chip: {
```
Comment (seanfeld) [inline app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: Okay sorry if I wasn't clear, none of these things should be done this way. We decided _new_ stuff uses styled components and all these are new. If you can't get something to work, like the tooltip, its fine to do that the other way but you should add a comment mentioning that. Chip and hyperlink should be done using styled components

Comment (sumengwang) [reply app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx]: made the changes

**Review by seanfeld — APPROVED** [review]

**Pull request merged**